designer/server/src/routes/forms/create.js

```diff
@@ -198,7 +200,13 @@ export default [
       // Submit new form metadata
       try {
         if (!result.error) {
```

Comment:
You'll have some type errors on the lines that follow

OpenID Connect (OIDC) doesn't guarantee `displayName` exists

So we'll need something like this with the current types:

```suggestion
        if (!result.error && credentials.user?.displayName) {
```

<br>

Or alternatively remove the question mark from the [`displayName?:` property](https://github.com/DEFRA/forms-designer/blob/2c556bb2a11eedae27d34f81fdd2aafd15b331c1/designer/server/src/types.ts#L60-L61)

Which only needs a simple check instead:

```suggestion
        if (!result.error && credentials.user) {
```

Reply:
Added NonNullable<> ask discussed.
Added an exception with @ts-expect-error for the user is not undefined case.

Reply:
Sorry @davidjamesstone

I'd prefer just tweaking the check then we can get this approved

We only use `@ts-expect-error` when we're purposely chucking incorrectly typed thing around

```suggestion
        if (!result.error && credentials.user) {
```